src/Components/Aspire.Qdrant/AspireQdrantExtensions.cs

```diff
@@ -0,0 +1,101 @@
+    /// </summary>
+    /// <param name="builder">The <see cref="IHostApplicationBuilder" /> to read config from and add services to.</param>
+    /// <param name="connectionName">The connection name to use to find a connection string.</param>
+    /// <param name="configureSettings">An optional method that can be used for customizing the <see cref="QdrantSettings"/>. It's invoked after the settings are read from the configuration.</param>
+    /// <remarks>Reads the configuration from "Aspire:Qdrant" section.</remarks>
+    /// <exception cref="InvalidOperationException">If required ConnectionString is not provided in configuration section</exception>
+    public static void AddQdrantClient(
+        this IHostApplicationBuilder builder,
+        string connectionName,
+        Action<QdrantSettings>? configureSettings = null)
+    {
+        AddQdrant(builder, DefaultConfigSectionName, configureSettings, connectionName, serviceKey: null);
+    }
+
+    /// <summary>
+    /// Registers <see cref="QdrantClient" /> as a singleton for given <paramref name="name" /> in the services provided by the <paramref name="builder"/>.
+    /// Configures logging and telemetry for the <see cref="QdrantClient" />.
+    /// </summary>
+    /// <param name="builder">The <see cref="IHostApplicationBuilder" /> to read config from and add services to.</param>
+    /// <param name="name">The connection name to use to find a connection string.</param>
+    /// <param name="configureSettings">An optional method that can be used for customizing the <see cref="QdrantSettings"/>. It's invoked after the settings are read from the configuration.</param>
+    /// <remarks>Reads the configuration from "Aspire:Qdrant" section.</remarks>
+    /// <exception cref="InvalidOperationException">If required ConnectionString is not provided in configuration section</exception>
+    public static void AddKeyedQdrantClient(
+        this IHostApplicationBuilder builder,
+        string name,
+        Action<QdrantSettings>? configureSettings = null)
+    {
+        AddQdrant(builder, $"{DefaultConfigSectionName}:{name}", configureSettings, connectionName: name, serviceKey: name);
+    }
+
+    private static void AddQdrant(
+        this IHostApplicationBuilder builder,
+        string configurationSectionName,
+        Action<QdrantSettings>? configureSettings,
+        string connectionName,
+        string? serviceKey)
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+
+        var settings = new QdrantSettings();
+        builder.Configuration.GetSection(configurationSectionName).Bind(settings);
+
+        if (builder.Configuration.GetConnectionString(connectionName) is string connectionString)
+        {
+            settings.ConnectionString = connectionString;
+        }
+
+        if (builder.Configuration[$"Parameters:{connectionName}-ApiKey"] is string apiKey)
+        {
+            settings.ApiKey = apiKey;
+        }
```

Comment:
This component needs a schema, it shouldn't use parameters, but instaed it should use something like `Qdrant:{connectionName}:ApiKey`

Reply:
@davidfowl it has a schema, see my comment on the other PR feedback as well -- agreed a better named one is better, but the component config schema's don't account for per-named resources (and their typed settings class)

Reply:
2 thoughts here:

1. For other components that look like this (need a URL and an API Key), we've defined a connection string format ourselves. See 
https://github.com/dotnet/aspire/blob/0d3d68e0d59c1e45d36484761f1516bcfc6e1bbd/src/Components/Aspire.Azure.AI.OpenAI/AzureOpenAISettings.cs#L50-L72

https://github.com/dotnet/aspire/blob/0d3d68e0d59c1e45d36484761f1516bcfc6e1bbd/src/Components/Aspire.Azure.Search.Documents/AzureSearchSettings.cs#L58-L80

I'm not sure we should invent yet another pattern for this component.

2. IMO, the correct fix/design here is https://github.com/dotnet/aspire/issues/1765.

Reply:
What is the suggested change here? In this case the ApiKey is not a part of the connection string, it's passed as an arg to the client lib.

Reply:
> In this case the ApiKey is not a part of the connection string, it's passed as an arg to the client lib.

That is no different than the 2 scenarios I linked to above. Here's where the ApiKeys get passed as an arg to the client lib for them:

https://github.com/dotnet/aspire/blob/126b1855ba08d7670747d59d2a8048d36d1c1c99/src/Components/Aspire.Azure.AI.OpenAI/AspireAzureOpenAIExtensions.cs#L118-L119

https://github.com/dotnet/aspire/blob/126b1855ba08d7670747d59d2a8048d36d1c1c99/src/Components/Aspire.Azure.Search.Documents/AspireAzureSearchExtensions.cs#L114

Reply:
Should IConnectionStrings move more generically from Azure.Component?

Reply:
> Should IConnectionStrings move more generically from Azure.Component?

No, the only reason it exists is because the base `AzureComponent` needs to call it on the Settings object. In your case, the `AspireQdrantExtensions` class can just call an internal method on the `QdrantSettings` class.

Reply:
Fixed in latest update